File: app/src/main/java/me/gabeg/sicksends/problem/type/SsHoldType.java
package me.gabeg.sicksends.problem.type;

/**
 * Types of climbing holds that a person may encounter.
 */
public enum SsHoldType
{

	CRIMP(1 << 0),
	HORN(1 << 1),
	JUG(1 << 2),
	PINCH(1 << 3),
	POCKET(1 << 4),
	SLOPER(1 << 5),
	SIDE_PULL(1 << 6),
	UNDERCLING(1 << 7),
	VOLUME(1 << 8);
	// Volume only for indoor

	/**
	 * Value associated with an enum.
	 */
	private final long mValue;

	/**
	 */
	private SsHoldType(long value)
	{
		this.mValue = value;
	}

	/**
	 * @return Value of the enum.
	 */
	public long getValue()
	{
		return this.mValue;
	}

}

File: app/src/main/java/me/gabeg/sicksends/problem/type/SsHoldType.kt
package me.gabeg.sicksends.problem.type

import java.util.*

/**
 * Types of climbing holds that a person may encounter.
 *
 * TODO: Volume only for indoor
 *
 * @param  value  Value associated with an enum.
 */
enum class SsHoldType(val value : Long)
{

	CRIMP(1 shl 0),
	HORN(1 shl 1),
	JUG(1 shl 2),
	PINCH(1 shl 3),
	POCKET(1 shl 4),
	SLOPER(1 shl 5),
	SIDE_PULL(1 shl 6),
	UNDERCLING(1 shl 7),
	VOLUME(1 shl 8);

	companion object
	{

		/**
		 * Create an empty EnumSet of this type.
		 *
		 * @return An empty EnumSet of this type.
		 */
		fun emptySet() : EnumSet<SsHoldType>
		{
			return EnumSet.noneOf(SsHoldType::class.java)
		}

	}

}